packages/manager/src/components/TagsInput/TagsInput.spec.js

```diff
@@ -2,89 +2,99 @@ const { navigateToStory } = require('../../../e2e/utils/storybook');
-        expect(inputError.getText()).toContain('Length must be 3-50 characters');
-      } catch (e) {
+  const inputValidation = (tagText) => {
+    $(tagInput).setValue(tagText);
+    $(tagOptions).waitForDisplayed(constants.wait.normal);
+    $$(tagOptions)[0].click();
+    try {
+      inputError.waitForVisible(constants.wait.normal);
+      expect(inputError.getText())
+        .withContext(`Incorrect error message`)
+        .toContain('Length must be 3-50 characters');
+    } catch (e) {
 
-      }
     }
+  }
 
-    beforeAll(() => {
-        navigateToStory(component, childStories[0]);
-    });
+  beforeAll(() => {
+    navigateToStory(component, childStories[0]);
+  });
 
-    it('there should be a tags input drop down with available tags', () => {
-        browser.waitForVisible(tagInput,constants.wait.normal);
-        $(tagInput).click();
-        browser.waitForVisible(tagOptions,constants.wait.normal);
-        const tags = $$(tagOptions).map(tag => tag.getText());
-        expect(tags).toEqual(['tag1','tag2','tag3','tag4']);
-    });
+  it('there should be a tags input drop down with available tags', () => {
+    $(selectBox).waitForDisplayed(constants.wait.normal);
+    $(selectBox).click();
+    $(tagOptions).waitForDisplayed(constants.wait.normal);
+    const tags = $$(tagOptions).map(tag => tag.getText());
+    expect(tags)
+      .withContext(`Incorrect tag names found`)
+      .toEqual(['tag1','tag2','tag3','tag4']);
+  });
 
-    it('an avialable tag can be selected', () => {
-        $$(tagOptions)[0].click();
-        browser.waitForVisible(selectedTags,constants.wait.normal);
-        expect($(selectedTags).getText()).toEqual('tag1');
-    });
+  it('an available tag can be selected', () => {
+    $$(tagOptions)[0].click();
+    $(selectedTags).waitForDisplayed(constants.wait.normal);
+    expect($(selectedTags).getText())
+      .withContext(`Incorrect text found`)
+      .toEqual('tag1');
+  });
 
-    it('multiple tags can be selected', () => {
-        $('input').click();
-        browser.waitForVisible(tagOptions,constants.wait.normal);
-        $$(tagOptions)[0].click();
-        browser.waitUntil(() => {
-            return $$(selectedTags).length === 2;
-        },constants.wait.normal);
-    });
+  it('multiple tags can be selected', () => {
+    $('input').click();
+    $(tagOptions).waitForDisplayed(constants.wait.normal);
+    $$(tagOptions)[0].click();
+    browser.waitUntil(() => {
+        return $$(selectedTags).length === 2;
+    },constants.wait.normal);
+  });
 
-    it('a single tag can be cleared by clicking the x icon on tag', () => {
-        $(`${selectedTags}`).$('..').$('svg').click();
-        browser.waitUntil(() => {
-            return $$(selectedTags).length === 1;
-        },constants.wait.normal);
-    });
+  it('a single tag can be cleared by clicking the x icon on tag', () => {
+    $(`${selectedTags}`).$('..').$('svg').click();
```

Comment:
not required, but it'd be great if we could target a more reliable selector here